pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,356 @@
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ (QSVT) circuit.
+
+    Given an :class:`~.Operator` :math:`U`, which block encodes the matrix :math:`A`, and a list of
+    projector-controlled phase shift operations :math:`\vec{\Pi}_\phi`, this template applies a
+    circuit for the quantum singular value transformation as follows.
+
+    .. note ::
+
+        This template allows users to define hardware-compatible block-encoding and
+        projector-controlled phase shift circuits. For a QSVT implementation that is
+        tailored to for simulators see :func:`~.qsvt` .
+
+
+    When the number of projector-controlled phase shifts is even (:math:`d` is odd), the QSVT
+    circuit is defined as:
+
+    .. math::
+
+        U_{QSVT} = \tilde{\Pi}_{\phi_1}U\left[\prod^{(d-1)/2}_{k=1}\Pi_{\phi_{2k}}U^\dagger
+        \tilde{\Pi}_{\phi_{2k+1}}U\right]\Pi_{\phi_{d+1}}.
+
+
+    And when the number of projector-controlled phase shifts is odd (:math:`d` is even):
+
+    .. math::
+
+        U_{QSVT} = \left[\prod^{d/2}_{k=1}\Pi_{\phi_{2k-1}}U^\dagger\tilde{\Pi}_{\phi_{2k}}U\right]
+        \Pi_{\phi_{d+1}}.
+
+    This circuit applies a polynomial transformation (:math:`Poly^{SV}`) to the singular values of
+    the block encoded matrix:
+
+    .. math::
+
+        \begin{align}
+             U_{QSVT}(A, \vec{\phi}) &=
+             \begin{bmatrix}
+                Poly^{SV}(A) & \cdot \\
+                \cdot & \cdot
+            \end{bmatrix}.
+        \end{align}
+
+    .. seealso::
+
+        :func:`~.qsvt` and `A Grand Unification of Quantum Algorithms <https://arxiv.org/pdf/2105.02859.pdf>`_.
+
+    Args:
+        UA (Operator): the block encoding circuit, specified as an :class:`~.Operator`,
+            like :class:`~.BlockEncode`
+        projectors (Sequence[Operator]): a list of projector-controlled phase
+            shifts that implement the desired polynomial
+        wires (Iterable): the wires the template acts on
+
+    Raises:
+        ValueError: if the input block encoding is not an operator
+
+    **Example**
+
+    To implement QSVT in a circuit, we can use the following method:
+
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> A = np.array([[0.1]])
+    >>> block_encode = qml.BlockEncode(A, wires=[0, 1])
+    >>> shifts = [qml.PCPhase(i + 0.1, dim=1, wires=[0, 1]) for i in range(3)]
+    >>> @qml.qnode(dev)
+    >>> def example_circuit():
+    ...    qml.QSVT(block_encode, shifts, wires=[0, 1])
+    ...    return qml.expval(qml.PauliZ(wires=0))
+
+    The resulting circuit implements QSVT.
+
+    >>> print(qml.draw(example_circuit)())
+    0: ─╭QSVT─┤  <Z>
+    1: ─╰QSVT─┤
+
+    To see the implementation details, we can expand the circuit:
+
+    >>> print(qml.draw(example_circuit,expansion_strategy='device')())
+    0: ─╭∏_ϕ(0.10)─╭BlockEncode(M0)─╭∏_ϕ(1.10)─╭BlockEncode(M0)†─╭∏_ϕ(2.10)─┤  <Z>
+    1: ─╰∏_ϕ(0.10)─╰BlockEncode(M0)─╰∏_ϕ(1.10)─╰BlockEncode(M0)†─╰∏_ϕ(2.10)─┤
```

Comment:
I cannot reproduce this print output, get this instead:
```
0: ─╭QSVT─┤  <Z>
1: ─╰QSVT─┤     
```

Reply:
Docstring will change in an upcoming commit.